ratespread-calculator/Jenkinsfile

```diff
@@ -24,7 +24,18 @@ volumes: [
     }
 
     println "DOCKER_TAG: ${env.DOCKER_TAG}, TAG_NAME: ${env.TAG_NAME}, gitbranch: ${gitBranch}, isDeployPR: ${isDeployPR}"
-     
+    
+    sh "git remote add upstream https://github.com/cfpb/hmda-platform.git"
+    sh "git fetch upstream master"
+
+    //Get list of changed files and end job if files changed aren't in relevant folders
+    folders=[]
+    def output = sh(returnStdout: true, script: "echo \$(git --no-pager diff upstream/master --name-only)").split().each{folders = folders + it.split('/')[0]}
+    if (!(gitBranch == 'master' || folders.contains('ratespread-calculator') || folders.contains('common'))) {
```

Comment:
@kgudel don't we also need an else clause that sets `currentBuild.result = FAILURE ` when the files changes don't match what is in that repo?

Reply:
This if clause actually addresses what happens when the files changed from a PR aren't contained in the relevant repo. The build is set to `SUCCESS` and then stopped so that the rest of the pipeline isn't run. If the files changed are in the repo the process flows to the pipeline code below as kind of an implicit else.

Reply:
ah missed the `!` in front. LGTM 👍